frontend/pipeline.yml

```
@@ -0,0 +1,300 @@
+      alias: docker
+  before_script:
+    - !reference [.dind-login]
+    - apk add --no-cache bash git jq wget
+    - docker pull ${CI_REGISTRY_IMAGE}:frontend-${CI_PIPELINE_ID}
+    - docker tag ${CI_REGISTRY_IMAGE}:frontend-${CI_PIPELINE_ID} ${FRONTEND_REPOSITORY}:pr
+  artifacts:
+    expire_in: 2w
+    paths:
+      - coverage
+      - screenshots
+      - logs
+      - junit
+    reports:
+      junit:
+        - junit/results.xml
+    when: always
+  tags:
+    - mender-qa-worker-generic-light
+
+test:frontend:acceptance:
+  extends: .template:test:frontend:acceptance
+  rules:
+    - if: $CI_COMMIT_REF_NAME !~ /^(staging|saas-[a-zA-Z0-9.]+)$/
+  script:
+    - GUI_REPOSITORY=$(pwd)/frontend SERVER_ROOT=$(pwd) ./frontend/tests/e2e_tests/run
+
+test:frontend:acceptance:enterprise:
+  extends: .template:test:frontend:acceptance
+  rules:
+    - if: $CI_COMMIT_BRANCH == "main"
+  variables:
+    TEST_ENVIRONMENT: enterprise
+  before_script:
+    - docker login -u $REGISTRY_MENDER_IO_USERNAME -p $REGISTRY_MENDER_IO_PASSWORD registry.mender.io
+  script:
+    - GUI_REPOSITORY=$(pwd)/frontend SERVER_ROOT=$(pwd) ./frontend/tests/e2e_tests/run --enterprise
+
+.template:publish:frontend:tests:
+  image: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/${NODE_IMAGE}
+  stage: publish
+  before_script:
+    - export CI_BUILD_REF=${CI_COMMIT_SHA}
+    - export CI_BUILD_REF_NAME=${CI_COMMIT_REF_NAME}
+    - export CI_MERGE_REQUEST_IID=${CI_COMMIT_BRANCH#pr_}
+    - export COVERALLS_PARALLEL=true
+    - export COVERALLS_SERVICE_JOB_ID=${CI_JOB_ID}
+    - export COVERALLS_SERVICE_NUMBER=${CI_PIPELINE_ID}
+    - apk add --no-cache git
+    - npm i -g coveralls
+  tags:
+    - hetzner-amd-beefy
+
+publish:frontend:tests:
+  extends: .template:publish:frontend:tests
+  rules:
+    - if: $CI_COMMIT_REF_NAME !~ /^(staging|saas-[a-zA-Z0-9.]+)$/
+  needs:
+    - test:frontend:unit
+  script:
+    - export COVERALLS_SERVICE_JOB_NUMBER=frontend-unit
+    - export COVERALLS_FLAG_NAME=frontend-unit
+    - coveralls < frontend/coverage/lcov.info
+
+publish:frontend:e2e-tests:
+  extends: .template:publish:frontend:tests
+  rules:
+    - if: $CI_COMMIT_BRANCH == "main"
+  needs:
+    - test:frontend:acceptance:enterprise
+  script:
+    - export COVERALLS_SERVICE_JOB_NUMBER=frontend-e2e
+    - export COVERALLS_FLAG_NAME=frontend-e2e
+    - sed -i -re 's/(^[SF:]+[../]+)(.*)$/SF:\2/' coverage/lcov.info
+    - coveralls < coverage/lcov.info
+
+publish:frontend:licenses:
+  stage: publish
+  image: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/docker:git
+  needs:
+    - job: build:frontend:docker
+      artifacts: true
+  services:
+    - name: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/docker:dind
+      alias: docker
+  script:
+    - docker load -i baseImage.tar
+    - docker run --rm --entrypoint "/bin/sh" -v $(pwd):/extract ${FRONTEND_REPOSITORY}:base -c "npm run disclaim && cp disclaimer.txt /extract/"
+    - docker image rm disclaimer
```

Following the suggestion above. This can be simplified significantly...
```suggestion
  image: ${CI_REGISTRY_IMAGE}/frontend-licenses-${CI_PIPELINE_IID}
  needs:
    - job: build:frontend:docker
      artifacts: false
  script:
    - npm run disclaim
```